coupled_laplace_problem/coupled_laplace_problem.cc

```diff
@@ -0,0 +1,674 @@
+    }
+
+  // TODO: Discuss position of this function: inside the if statement leads to a
+  // precice error, because no initial data is required, but it needs to be
+  // initialized
+  precice.initializeData();
+
+  // read initial readData from preCICE if required for the first time step
+  // FIXME: Data is already exchanged here
+  if (precice.isReadDataAvailable())
+    {
+      precice.readBlockScalarData(read_data_id,
+                                  n_interface_nodes,
+                                  interface_nodes_ids.data(),
+                                  read_data.data());
+
+      // This is the opposite direction as above. See comment there.
+      format_precice_to_deal(boundary_data);
+    }
+}
+
+
+// The function `advance()` is called in the main time loop after the
+// computation in each time step in the individual solver has finished. Here,
+// coupling data is passed to preCICE and obtained from other participants. In
+// case of the simplified unidirectional coupling, we just obtain data from our
+// dummy participant.
+template <int dim, typename ParameterClass>
+void
+Adapter<dim, ParameterClass>::advance(
+  std::map<types::global_dof_index, double> &boundary_data,
+  const double                               computed_timestep_length)
+{
+  // This is essentially the same as during initialization. We have already all
+  // IDs and just need to convert our obtained data to the preCICE compatible
+  // `write_data` vector. Since we deal here with a unidirectional example,
+  // where the Laplace solver doesn't write any data, we don't need to transform
+  // any data and the following code block is not accessed, i.e.
+  // `precice.hasData()` will return false.
+  // TODO: Should we remove this block here? No data is written, but it would
+  // illustrate the concept.
+  if (precice.hasData(write_data_name, mesh_id))
+    if (precice.isWriteDataRequired(computed_timestep_length))
+      {
+        precice.writeBlockScalarData(write_data_id,
+                                     n_interface_nodes,
+                                     interface_nodes_ids.data(),
+                                     write_data.data());
+      }
+
+  // Here, we specify the computed time step length and pass it to preCICE
+  precice.advance(computed_timestep_length);
+
+  // As a next step, we obtain data, i.e. the boundary condition, from another
+  // participant. We have already all IDs and just need to convert our obtained
+  // data to the deal.II compatible 'boundary map' , which is done in the
+  // format_deal_to_precice function. All this is of course only done in
+  // case write data is required.
+  if (precice.isReadDataAvailable())
+    {
+      precice.readBlockScalarData(read_data_id,
+                                  n_interface_nodes,
+                                  interface_nodes_ids.data(),
+                                  read_data.data());
+
+      format_precice_to_deal(boundary_data);
+    }
```

Comment:
```suggestion
    precice.readBlockScalarData(read_data_id,
                                n_interface_nodes,
                                interface_nodes_ids.data(),
                                read_data.data());

    format_precice_to_deal(boundary_data);
```

We could actually even do this.